.github/actions/changelog-check/action.yml

```diff
@@ -0,0 +1,67 @@
+name: 'Changelog Check'
+description: 'Check if changelog is updated for the changed packages'
+inputs:
+  github_token:
+    description: 'GitHub token'
+    required: true
+  pr_number:
+    description: 'Pull request number'
+    required: true
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0  # Fetch full history to access all commits
+
+    - name: Fetch PR labels
+      id: fetch_labels
+      run: |
+        PR_LABELS=$(curl -s -H "Authorization: token ${{ inputs.github_token }}" \
+          "https://api.github.com/repos/${{ github.repository }}/issues/${{ inputs.pr_number }}/labels" | \
+          jq -r '.[].name')
+        echo "PR_LABELS=${PR_LABELS}" >> $GITHUB_ENV
+      shell: bash
+
+    - name: Get Base Branch SHA
+      id: get_base_sha
+      run: |
+        BASE_SHA=$(curl -s -H "Authorization: token ${{ inputs.github_token }}" \
+          "https://api.github.com/repos/${{ github.repository }}/pulls/${{ inputs.pr_number }}" | \
+          jq -r '.base.sha')
+        echo "BASE_SHA=${BASE_SHA}" >> $GITHUB_ENV
+      shell: bash
+
+    - name: Debugging Info
+      run: |
+        echo "Base SHA: ${{ env.BASE_SHA }}"
+        echo "Current SHA: ${{ github.sha }}"
+        echo "PR labels: $PR_LABELS"
+      shell: bash
+
+    - name: Check if changelog is updated
+      run: |
+        if [ -z "${{ env.BASE_SHA }}" ] || [ -z "${{ github.sha }}" ]; then
+          echo "Error: SHA values are missing or invalid."
+          exit 1
+        fi
+
+        CHANGED_FILES=$(git diff --name-only ${{ env.BASE_SHA }} ${{ github.sha }})
+        echo "Changed files: $CHANGED_FILES"
+        
+        PUBLIC_PACKAGES=("commerce-sdk-react" "pwa-kit-create-app" "pwa-kit-dev" "pwa-kit-react-sdk" "pwa-kit-runtime" "template-retail-react-app")
+        SKIP_CHANGELOG_LABEL="skip changelog" 
+
+        for PACKAGE in "${PUBLIC_PACKAGES[@]}"; do
+          if echo "$CHANGED_FILES" | grep -q "^packages/$PACKAGE/"; then
+            if ! echo "$CHANGED_FILES" | grep -q "^packages/$PACKAGE/changelog.md"; then
+              if ! echo "$PR_LABELS" | grep -q "$SKIP_CHANGELOG_LABEL"; then
```

Comment:
NIT: Could we not exit early with a success if the skip changelog label is present? E.g. If right after we fetch the labels see if we are skipping then exit(0) or whatever to signal a success